RGPosadas/LitterRally
# TASK-8: Setup Media Backend

**Task Description**
This PR addresses #36.

**Note about GraphQL decorators and code coverage**
If you see ```/* istanbul ignore next */``` on any of the GraphQL classes, it is to ignore them from the code coverage since the decorators take in anonymous functions.  

**Testing Steps Using Postman**
1. Make sure to resolve all your dependencies with `npm i`
1. Run the application using ```nx serve mull-api```.
1. Open the Postman application
1. Select POST from the dropdown
1. Enter URL to GraphQL server `http://localhost:3333/graphql`
1. Select the `Body` tab and `form-data`
1. Fill out the `Body` tab as follows:

| Key | Value |
| ------------- | ------------- |
| operations | {"query":"mutation UploadFile($file: Upload!) {uploadFile(file:$file)}", "variables": {"file": null}}  |
| map | {"0": ["variables.file"]}  |
| 0 | [File ▾] (select your file to upload) |

8. Observe that the image is uploaded within the `apps/mull-api/uploads` folder and stored in the database
